platform/openide.util.lookup/test/unit/src/org/openide/util/LookupTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.openide.util;
+
+import java.util.Optional;
+import org.junit.Test;
+import org.netbeans.junit.NbTestCase;
+
+/**
+ *
+ * @author Hector Espert
+ */
+public class LookupTest extends NbTestCase {
+    
+    private Lookup lookup;
+
+    public LookupTest(String name) {
+        super(name);
+    }
+    
+    @Override
+    public void setUp() {
+        lookup = new LookupImpl();
+    }
+
+    @Test
+    public void testLookupOptional() {
+        Optional<Integer> optionalInteger = lookup.lookupOptional(Integer.class);
+        assertFalse(optionalInteger.isPresent());
+        
+        Optional<String> optionalString = lookup.lookupOptional(String.class);
```

Comment:
This test is fine, but there are many implementations of `Lookup` and we have to test them all. There is a base "TCK-like" test `AbstractLookupBaseHid` - if you add a test method in there, it is going to be executed against all the known implementations of lookup - `ProxyLookup`, `AbstractLookup`, etc.